### PR TITLE
Fixes #16003: Add shared-folder get to API doc

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/16003

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: src/relay-api/endpoints/sharedFolder.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ This file describes shared-folder REST API
     @apiName sharedFolderHash
     @apiGroup Shared folder
     
-    @apiDescription Allows checking for a file content hash, typically to check if an update is available. If the file hash matches the computed one, a `304` code is returned. If the file exists but does not match the hash (or the hash is absent), a `200` code id returned. Finally, if the file does not exist, a `404` code is returned.
+    @apiDescription Allows checking for a file content hash, typically to check if an update is available. If the file hash matches the computed one, a `304` code is returned. If the file exists but does not match the hash (or the hash is absent), a `200` code id returned. Finally, if the file does not exist, a `404` code is returned. Howerver, the returned `content-length` and `content-type` do not match those of the target file (accessible with a `GET`). The client has to authenticate with TLS client authentication using the agent certificate and key, the access is only granted to nodes that are managed by the relay.
```
Comment (ncharles) [inline src/relay-api/endpoints/sharedFolder.md]: i don't understand the However part (and there's a typo in however)

Comment (amousset) [reply src/relay-api/endpoints/sharedFolder.md]: The `HEAD` headers should be the same as what the same request with `GET` would give, but we don't really do that. I'll rephrase it.

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Pull request merged**